frontend/src/Onboarding/TransferableCreditScreen.tsx

```diff
@@ -0,0 +1,182 @@
+import { Grid, Paper } from "@material-ui/core";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { useLocation } from "react-router-dom";
+import { TransferableExam, TransferableExamGroup } from "../../../common/types";
+import { setExamCredits } from "../state/actions/userActions";
+import {
+  MainTitleText,
+  OnboardingSelectionTemplate,
+  SelectableCourse,
+  TitleText,
+} from "./GenericOnboarding";
+import { APExamGroups2020To2021 } from "../../../common/ap_exams";
+import { IBExamGroups2020To2021 } from "../../../common/ib_exams";
+
+interface TransferableExamGroupComponentProps {
+  readonly transferableExamGroup: TransferableExamGroup;
+  readonly selectedTransferableExams: Array<TransferableExam>;
+  readonly setSelectedTransferableExams: (
+    transferableExams: TransferableExam[]
+  ) => void;
+}
+
+interface TransferableExamComponentProps {
+  readonly transferableExam: TransferableExam;
+  readonly selectedTransferableExams: Array<TransferableExam>;
+  readonly setSelectedTransferableExams: (
+    transferableExams: TransferableExam[]
+  ) => void;
+}
+
+interface TransferableExamGroupsComponentProps {
+  readonly transferableExamGroups: TransferableExamGroup[];
+  readonly selectedTransferableExams: Array<TransferableExam>;
+  readonly keyPrefix: string;
+  readonly setSelectedTransferableExams: (
+    transferableExams: TransferableExam[]
+  ) => void;
+}
+
+const TransferableExamComponent: React.FC<
+  TransferableExamComponentProps
+> = props => {
+  const addCourseToSelected = () => {
+    const newSelectedTransferableExams: Array<TransferableExam> = [
+      ...props.selectedTransferableExams,
+      props.transferableExam,
+    ];
+    props.setSelectedTransferableExams(newSelectedTransferableExams);
+  };
+
+  const removeCourseFromSelected = () => {
+    const newSelectedTransferableExams: Array<
+      TransferableExam
+    > = props.selectedTransferableExams.filter(
+      (transferableExam: TransferableExam) =>
+        transferableExam.name !== props.transferableExam.name
+    );
+    props.setSelectedTransferableExams(newSelectedTransferableExams);
+  };
+
+  const onChecked = (e: any): void => {
+    const checked = e.target.checked;
+    if (checked) {
+      addCourseToSelected();
+    } else {
+      removeCourseFromSelected();
+    }
+  };
+
+  return (
+    <SelectableCourse
+      key={`${props.transferableExam.type}-${props.transferableExam.name}`}
+      onChange={onChecked}
+      courseText={props.transferableExam.name}
+    />
+  );
+};
+
+const TransferableExamGroupComponent: React.FC<
```

Comment:
Only thing is maybe add some comments for each of the Components, since the naming is super similar and it makes it a bit confusing